jdaviz/configs/cubeviz/plugins/tests/test_parsers.py

```diff
@@ -151,5 +168,49 @@ def test_spectrum1d_parse(spectrum1d, cubeviz_helper):
 
 
 def test_numpy_cube(cubeviz_helper):
+    arr = np.ones(24).reshape((2, 3, 4))  # z, y, x
+
+    with pytest.raises(TypeError, match='Data type must be one of'):
+        cubeviz_helper.load_data(arr, data_type='foo')
+
+    cubeviz_helper.load_data(arr)
+    cubeviz_helper.load_data(arr << u.nJy, data_label='uncert_array', data_type='uncert',
+                             override_cube_limit=True)
+
+    with pytest.raises(RuntimeError, match='Only one cube'):
+        cubeviz_helper.load_data(arr, data_type='mask')
+
+    assert len(cubeviz_helper.app.data_collection) == 2
+
+    # Check context of first cube.
+    data = cubeviz_helper.app.data_collection[0]
+    flux = data.get_component('flux')
+    assert data.label == 'Unknown array'
+    assert data.shape == (4, 3, 2)  # x, y, z
+    assert isinstance(data.coords, PaddedSpectrumWCS)
+    assert flux.units == 'ct'
+
+    # Check context of second cube.
+    data = cubeviz_helper.app.data_collection[1]
+    flux = data.get_component('flux')
+    assert data.label == 'uncert_array'
+    assert data.shape == (4, 3, 2)  # x, y, z
+    assert isinstance(data.coords, PaddedSpectrumWCS)
+    assert flux.units == 'nJy'
+
+
+def test_invalid_data_types(cubeviz_helper):
+    with pytest.raises(FileNotFoundError, match='Could not locate file'):
+        cubeviz_helper.load_data('does_not_exist.fits')
+
+    with pytest.raises(NotImplementedError, match='Unsupported data format'):
+        cubeviz_helper.load_data(WCS(naxis=3))
+
+    with pytest.raises(NotImplementedError, match='Unsupported data format'):
+        cubeviz_helper.load_data(Spectrum1D(flux=np.ones((2, 2)) * u.nJy))
+
+    with pytest.raises(NotImplementedError, match='Unsupported data format'):
+        cubeviz_helper.load_data(np.ones((2, 2)))
```

Comment:
In a previous life of this PR over at #1040 , I also made 2D image loading possible but I didn't do it this time because I think we did some unholy WCS hacks to get the 2D moment map to work with the cube, so I don't want to go back to that rabbit hole.